src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx

```diff
@@ -20,6 +20,512 @@ export function getStaticProps(context) {
+
+```typescript
+import {
+	confirmSignIn,
+	confirmSignUp,
+	resetPassword,
+	signIn,
+} from 'aws-amplify/auth';
+
+const { nextStep } = await signIn({
+	username: 'hello@mycompany.com',
+	password: 'hunter2',
+});
+
+if (
+	nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_SMS_CODE' ||
+	nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_EMAIL_CODE' ||
+	nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_TOTP_CODE'
+) {
+	// collect OTP from user
+	await confirmSignIn({
+		challengeResponse: '123456',
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_MFA_SELECTION') {
+	// present nextStep.allowedMFATypes to user
+	// collect user selection
+	await confirmSignIn({
+		challengeResponse: 'EMAIL', // 'EMAIL', 'SMS', or 'TOTP'
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_MFA_SETUP_SELECTION') {
+	// present nextStep.allowedMFATypes to user
+	// collect user selection
+	await confirmSignIn({
+		challengeResponse: 'EMAIL', // 'EMAIL' or 'TOTP'
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_EMAIL_SETUP') {
+	// collect email address from user
+	await confirmSignIn({
+		challengeResponse: 'hello@mycompany.com',
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_TOTP_SETUP') {
+	// present nextStep.totpSetupDetails.getSetupUri() to user
+	// collect OTP from user
+	await confirmSignIn({
+		challengeResponse: '123456',
+	});
+}
+
+if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
+	// collect custom challenge answer from user
+	await confirmSignIn({
+		challengeResponse: 'custom-challenge-answer',
+	});
+}
+
+if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_NEW_PASSWORD_REQUIRED') {
+	// collect new password from user
+	await confirmSignIn({
+		challengeResponse: 'new-password',
+	});
+}
+
+if (nextStep.signInStep === 'RESET_PASSWORD') {
+	// initiate reset password flow
+	await resetPassword({
+		username: 'username',
+	});
+}
+
+if (nextStep.signInStep === 'CONFIRM_SIGN_UP') {
+	// user was not confirmed during sign up process
+	// if user has confirmation code, invoke `confirmSignUp` api
+	// otherwise, invoke `resendSignUpCode` to resend the code
+	await confirmSignUp({
+		username: 'username',
+		confirmationCode: '123456',
+	});
+}
+
+if (nextStep.signInStep === 'DONE') {
+	// signin complete
+}
+```
+
+## Confirm signin with SMS MFA
+
+If the next step is `CONFIRM_SIGN_IN_WITH_SMS_CODE`, Amplify Auth has sent the user a random code over SMS and is waiting for the user to verify that code. To handle this step, your app's UI must prompt the user to enter the code. After the user enters the code, pass the value to the `confirmSignIn` API.
+
+<Callout>
+
+The result includes `codeDeliveryDetails` with additional information about the code delivery, such as the partial phone number of the SMS recipient, which can be used to prompt the user on where to look for the code.
+
+</Callout>
+
+```ts
+async function handleSignInResult(result: SignInOutput) {
```

Comment:
missing imports for the type and functions used

Reply:
[added imports here](https://github.com/aws-amplify/docs/pull/8023/commits/14ec2ed85834f7cd752cb9d2e7cdf6cb00c10c6a)